src/sentry/api/event_search.py

```diff
@@ -2021,6 +2047,26 @@ def result_type_fn(function_arguments, parameter_values):
             result_type_fn=reflective_result_type(),
             default_result_type="duration",
         ),
+        Function(
+            "compare_aggregate",
+            required_args=[
+                FunctionArg("aggregate_alias"),
+                ConditionArg("condition"),
+                NumberRange("value", 0, None),
+            ],
+            optional_args=[
+                with_default("pass", StringArg("pass")),
+                with_default("fail", StringArg("fail")),
+            ],
+            aggregate=[
+                # snuba json syntax isn't compatible with this query here
+                # can't be a column, since we want to use this with aggregates
+                "if({condition}({aggregate_alias},{value}),{pass},{fail})",
```

Comment:
Do we need to ensure that the aggregate_alias is safe? Right now this looks like we could be open to naughty queries.

Reply:
going to use the same regex as #22129 for this, also updated the other functions that string substitute too